src/python_testing/TC_CADMIN_1_22_24.py

```diff
@@ -0,0 +1,171 @@
+        await self.th1.SendCommand(nodeid=self.dut_node_id, endpoint=0, payload=revokeCmd, timedRequestTimeoutMs=6000)
+        # The failsafe cleanup is scheduled after the command completes, so give it a bit of time to do that
+        sleep(1)
+
+        self.step(4)
+        window_status = await self.get_window_status()
+        if window_status != Clusters.AdministratorCommissioning.Enums.CommissioningWindowStatusEnum.kWindowNotOpen:
+            asserts.fail("Commissioning window is expected to be closed, but was found to be open")
+
+        self.step(5)
+        try:
+            await self.th1.OpenCommissioningWindow(
+                nodeid=self.dut_node_id, timeout=901, iteration=10000, discriminator=self.discriminator, option=1)
+
+        except ChipStackError as e:
+            # Since we provided 901 seconds as the timeout duration,
+            # we should not be able to open comm window as duration is too long.
+            asserts.assert_equal(e.err,  0x00000585,
+                                 "Expected to error as we provided failure value for opening commissioning window")
+
+        self.step(6)
+        window_status2 = await self.get_window_status()
+        if window_status2 != Clusters.AdministratorCommissioning.Enums.CommissioningWindowStatusEnum.kWindowNotOpen:
+            asserts.fail("Commissioning window is expected to be closed, but was found to be open")
+
+    def pics_TC_CADMIN_1_24(self) -> list[str]:
+        return ["CADMIN.S"]
+
+    def steps_TC_CADMIN_1_24(self) -> list[TestStep]:
+        return [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "TH_CR1 opens a commissioning window on DUT_CE using ECM with a value of 180 seconds",
+                     "DUT_CE opens its Commissioning window to allow a second commissioning"),
+            TestStep(3, "TH_CR1 sends an RevokeCommissioning command to the DUT"),
+            TestStep(4, "TH_CR1 reads the window status to verify the DUT_CE window is closed",
+                     "DUT_CE windows status shows the window is closed"),
+            TestStep(5, "TH_CR1 opens a commissioning window on DUT_CE using ECM with a value of 179 seconds",
+                     "DUT_CE does not open its Commissioning window to allow a second commissioning. DUT_CE shows 'Failed to open commissioning window. Global status 0x85'"),
+            TestStep(6, "TH_CR1 reads the window status to verify the DUT_CE window is closed",
+                     "DUT_CE windows status shows the window is closed"),
+        ]
+
+    @async_test_body
+    async def test_TC_CADMIN_1_24(self):
+        self.step(1)
+        self.th1 = self.default_controller
+        self.discriminator = random.randint(0, 4095)
+
+        self.step(2)
+        await self.th1.OpenCommissioningWindow(
+            nodeid=self.dut_node_id, timeout=180, iteration=10000, discriminator=self.discriminator, option=1)
+
+        self.step(3)
```

Comment:
Step 3 for TC_CADMIN_1_24 in https://github.com/CHIP-Specifications/chip-test-plans/pull/4673/ says to wait 181 seconds for this to expire. It doesn't say to issue a revoke command. Should the testplan be updated or should you be waiting 181s?

Reply:
Hi Terence,
I have updated this to using the timeout of 180 seconds instead of the revoke commissioning command for step 3. Please note that where your comment is, is for TC_CADMIN_1_24 and the PR you mentioned was for TC_CADMIN_1_22. 
Note that there is an issue currently for when attempting to communicate with the endpoint after the timeout of the commissioning window, which I had filed an issue for already:
https://github.com/project-chip/connectedhomeip/issues/35983
We have a workaround already in place to get past that part until the issue is resolved, which you will notice in the latest change to the test module, which was is currently to just try again the get_window_status function.

Reply:
Ah that is a really good point, wish that it would allow for comments anywhere in the PR files, unfortunately, they block that. Thank you for the comment, I have updated as mentioned above to using the 180 second timeout during TC_CADMIN_1_24 and have continued with the revoke command for TC_CADMIN_1_22 as we just didn't want either the test harness or the testers having to wait the full 900 seconds before proceeding to the end of that test, which is due to the current issue #35983, we also have some other tests in CADMIN that perform the wait for timeout as well.